src/components/listBox/tests/ListBox.spec.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import { mount, ReactWrapper, shallow } from 'enzyme';
+import * as _ from 'underscore';
+import { IListBoxProps, ListBox } from '../ListBox';
+import { ItemBox } from '../../itemBox/ItemBox';
+
+describe('ListBox', () => {
+
+  let listBoxComponent: ReactWrapper<IListBoxProps, any>;
+
+  const defaultProps: IListBoxProps = {
+    items: [
+      { value: 'test' },
+      { value: 'test1' },
+      { value: 'test2' },
+      { value: 'test3' },
+      { value: 'test4' },
+      { value: 'test5' },
+      { value: 'test6' },
+      { value: 'test7' },
+    ],
+  };
+
+  it('should render without errors', () => {
+    expect(() => {
+      shallow(<ListBox {...defaultProps} />);
+    }).not.toThrow();
+  });
+
+  describe('<BoxItem /> with default props', () => {
+
+    beforeEach(() => {
+      listBoxComponent = mount(
+        <ListBox {...defaultProps} />,
+        { attachTo: document.getElementById('App') },
+      );
+    });
+
+    it('should render with the box-item class', () => {
+      expect(listBoxComponent.find(ItemBox).length).toBe(defaultProps.items.length);
+    });
+  });
+
+  describe('<BoxItem /> with custom props', () => {
+
+    const renderListBox = (props: Partial<IListBoxProps> = {}) => {
+      listBoxComponent = mount(
+        <ListBox {..._.defaults(props, defaultProps) } />,
+        { attachTo: document.getElementById('App') },
+      );
+    };
+
+    it('should render items with events on onOptionClick', () => {
+      const onOptionClick: jasmine.Spy = jasmine.createSpy('onOptionClick');
+      renderListBox({
+        onOptionClick,
+      });
+
+      listBoxComponent.find(ItemBox).node.handleOnOptionClick({ target: 'target' });
+      expect(onOptionClick).toHaveBeenCalled();
+    });
+
+    it('should show the no result <BoxItem/> if the items array is empty', () => {
+      renderListBox({
+        items: [],
+      });
+
+      expect(listBoxComponent.find(ItemBox).length).toBe(1);
```

Comment:
The expect is not really coherent with the test describe. Maybe you should check the text or classes instead?

Reply:
you should also test what is inside the box and check if it contains the string "No items"

Reply:
test added for the no result box